alembic/versions/20240821_7a2fcaac8b63_add_loan_identifier_column_to_playtime_tables.py

```diff
@@ -0,0 +1,103 @@
+"""Add loan_identifier column to playtime tables.
+
+Revision ID: 7a2fcaac8b63
+Revises: 7ba553f3f80d
+Create Date: 2024-08-21 23:23:48.085451+00:00
+
+"""
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.orm.session import Session
+
+# revision identifiers, used by Alembic.
+revision = "7a2fcaac8b63"
+down_revision = "7ba553f3f80d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    session = Session(bind=op.get_bind())
+    conn = session.connection()
+
+    op.add_column(
+        "playtime_entries",
+        sa.Column("loan_identifier", sa.String(length=50), nullable=False, default=""),
+    )
+
+    op.drop_constraint(
+        "unique_playtime_entry",
+        "playtime_entries",
+        type_="unique",
+    )
+
+    op.create_unique_constraint(
+        "unique_playtime_entry",
+        "playtime_entries",
+        [
+            "tracking_id",
+            "identifier_str",
+            "collection_name",
+            "library_name",
+            "loan_identifier",
+        ],
+    )
+
+    op.add_column(
+        "playtime_summaries",
+        sa.Column("loan_identifier", sa.String(length=50), nullable=False, default=""),
+    )
+
+    op.drop_constraint(
+        "unique_playtime_summary",
+        "playtime_summaries",
+        type_="unique",
+    )
+
+    op.create_unique_constraint(
+        "unique_playtime_summary",
+        "playtime_summaries",
+        [
+            "timestamp",
+            "identifier_str",
+            "collection_name",
+            "library_name",
+            "loan_identifier",
+        ],
+    )
+
+
+def downgrade() -> None:
+    op.drop_constraint(
+        "unique_playtime_entry",
+        "playtime_entries",
+        type_="unique",
+    )
+
+    op.drop_column("playtime_entries", "loan_identifier")
+
+    op.create_unique_constraint(
+        "unique_playtime_entry",
+        "playtime_entries",
+        [
+            "tracking_id",
+            "timestamp",
```

Comment:
The appearance of `timestamp` in this downgrade is unexpected, since the previous index did not include it...
```
__table_args__ = (
        UniqueConstraint(
            "tracking_id",
            "identifier_str",
            "collection_name",
            "library_name",
            name="unique_playtime_entry",
        ),
    )```

Reply:
Very good catch.